### DISCUSSION TITLE
Remove RDoc integration from RubyGems

### DISCUSSION THREAD

**Discussion by voxik** [ruby/rubygems#6021] [Ideas]
Trying to improve the situation with gems documentation on Fedora, where there are [long time concerns](https://bugzilla.redhat.com/show_bug.cgi?id=1224715) about bundling fonts, code duplication etc, I am looking into the details how this integration works. This brings me to the idea of removing the RDoc integration from RubyGems. With my current level of understanding, it seems that it would be much better, if the RDoc integration was done via RubyGems plugin and it would live either in RDoc or in some separate rdoc-rubygems integration gem.

Please note that I am not suggesting to drop this feature altogether. I just think that RDoc support is unnecessarily hacked into RubyGems while it seems that it could happily use RubyGems standard plugin interface. The benefit would be simpler RubyGems code base.

**Comment by simi** [comment]
:thinking: Like a plugin distributed by default?

**Comment by voxik** [comment]
If that was part of RDoc, then it would be distributed by default as part of RDoc. However, I am quite sure that the plugins are not properly supported for default gems.

If it was separate gem, then this would need to be distributed via similar means as RDoc (with similar issues).

**Comment by kou** [comment]
Plugin based implementation: https://github.com/ruby/rdoc/pull/1171

**Comment by hsbt** [comment]
This is done. RDoc registers the install hook itself as a RubyGems plugin from 6.9.0, added in https://github.com/ruby/rdoc/pull/1171, and https://github.com/ruby/rubygems/pull/8340 dropped the RubyGems side. What is left here is a compatibility shim for older RDoc, and it goes away once no supported Ruby ships RDoc below 6.9.0.
